org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/PositionerRequestHandler.java

```diff
@@ -1,88 +1,88 @@
-			if (request.getPositionType()==PositionRequestType.SET && request.getPosition()!=null) {
-				boolean ok = positioner.setPosition(request.getPosition());
-				if (!ok) throw new EventException("Internal Error: setPosition() did not return ok!");
-				
-			} else if (request.getPositionType()==PositionRequestType.ABORT) {
-				positioner.abort();
-			}else if (request.getPositionType()==PositionRequestType.CLOSE) {
-				positioner.close();
-			}
-			// Return the current position.
-			request.setPosition(positioner.getPosition());
-			return request;
-			
-		} catch(ScanningException | InterruptedException ne) {
-			throw new EventException("Cannot connect to positioner!", ne);
-		}
-	}
-
-	private IPositioner getPositioner(PositionerRequest request) throws ScanningException {
-		final String id = request.getUniqueId();
-		if (positioners.containsKey(id) && positioners.get(id).get()!=null) {
-			return positioners.get(id).get();
-		}
-
-		IPositioner positioner = dservice.createPositioner();
-		positioners.put(request.getUniqueId(), new SoftReference<IPositioner>(positioner));
-		return positioner;
-	}
-
-}
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.api.device;
+
+import java.lang.ref.Reference;
+import java.lang.ref.SoftReference;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.scanning.api.event.EventException;
+import org.eclipse.scanning.api.event.core.IPublisher;
+import org.eclipse.scanning.api.event.core.IRequestHandler;
+import org.eclipse.scanning.api.event.scan.PositionRequestType;
+import org.eclipse.scanning.api.event.scan.PositionerRequest;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.event.IPositioner;
+
+public class PositionerRequestHandler implements IRequestHandler<PositionerRequest>{
+	
+	private static Map<String, Reference<IPositioner>> positioners;
+
+	private IRunnableDeviceService        dservice;
+	private PositionerRequest             bean;
+	private IPublisher<PositionerRequest> publisher;
+
+	public PositionerRequestHandler(IRunnableDeviceService        dservice, 
+			                  PositionerRequest             bean, 
+			                  IPublisher<PositionerRequest> statusNotifier) {
+
+		this.dservice    = dservice;
+		this.bean        = bean;
+		this.publisher   = statusNotifier;
+		if (positioners==null) positioners = new HashMap<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this assignment of "positioners". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3010)